### PR TITLE
73 Mobile footer

### PR DESCRIPTION
**Pull request by mindfulme**
#### What's this PR do?
Adds mobile footer
#### Where should the reviewer start?
At MobileFooter.vue
#### How should this be manually tested?
the component is placed on the components section
#### Any background context you want to provide?
The component appears on mobile screens only (576px or less), adds several buttons that can be pressed
#### What are the relevant tickets?
[#73](https://github.com/bitshares/bitshares-community-ui/issues/73)
#### Screenshots (if appropriate)
screen shot 2018-10-21 at 11 03 21

#### URL
Provide a link to your deploy from http://staging-ui.trusty.apasia.tech:8080/
http://staging-ui.trusty.apasia.tech:8080/73-mobile-footer
#### Questions:
What can be improved?

### CODE REVIEW

File: src/components/MobileFooter/MobileFooter.vue [hunk 1]
```diff
@@ -0,0 +1,97 @@
+<template>
+  <div class="mobile-footer">
+    <div
+      v-for="(item, index) in items"
+      :key="index"
+      :style="{ width: itemWidth + '%' }"
+      :class="activeIndex === index ? 'active' : ''"
+      class="footer-item"
+      @click="$emit('itemClick')"
+      @itemClick="handleClick(index)"
```
Comment (roma219) [inline src/components/MobileFooter/MobileFooter.vue]: what is `@itemClick` ?

Comment (roma219) [reply src/components/MobileFooter/MobileFooter.vue]: Why so many emits?

File: src/components/MobileFooter/MobileFooter.vue [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    >
+      <svgicon
+        class="icon"
+        name="paste"
```
Comment (roma219) [inline src/components/MobileFooter/MobileFooter.vue]: icon name should be gotten from item

File: src/components/MobileFooter/MobileFooter.vue [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+      /><br>
+      <div class="text">{{ item }}</div>
```
Comment (roma219) [inline src/components/MobileFooter/MobileFooter.vue]: item should be an object with a structure `{ name, title, icon }`

File: src/components/MobileFooter/MobileFooter.vue [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+    handleClick(index) {
+      this.activeIndex = index
+      this.$emit('change', this.items[index])
+    }
+  }
+}
+</script>
+
+<style>
+@media (min-width: config('screens.sm')) {
```
Comment (roma219) [inline src/components/MobileFooter/MobileFooter.vue]: No media queries - https://tailwindcss.com/docs/

File: src/components/MobileFooter/MobileFooter.vue [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+  @apply m-0;
+}
+
+.footer-item:hover {
+     background-color: config('colors.mobile-footer-hover');
+     opacity: 1;
+}
+
+.icon {
+  width:33px;
```
Comment (roma219) [inline src/components/MobileFooter/MobileFooter.vue]: no pixels

**Review by roma219 — CHANGES_REQUESTED** [review]
Needs changes

Doesn't look like design:
image

File: src/components/MobileFooter/MobileFooter.vue (hunk 5 above)
Comment (mindfulme) [reply src/components/MobileFooter/MobileFooter.vue]: got it

File: src/components/MobileFooter/MobileFooter.vue (hunk 4 above)
Comment (mindfulme) [reply src/components/MobileFooter/MobileFooter.vue]: ok, it is better without `@media`

File: src/components/MobileFooter/MobileFooter.vue (hunk 3 above)
Comment (mindfulme) [reply src/components/MobileFooter/MobileFooter.vue]: okay

File: src/components/MobileFooter/MobileFooter.vue (hunk 2 above)
Comment (mindfulme) [reply src/components/MobileFooter/MobileFooter.vue]: as a field, ok

File: src/components/MobileFooter/MobileFooter.vue (hunk 1 above)
Comment (mindfulme) [reply src/components/MobileFooter/MobileFooter.vue]: It is not needed anymore. Okay, I will refactor

File: src/components/MobileFooter/MobileFooter.vue (hunk 4 above)
Comment (mindfulme) [reply src/components/MobileFooter/MobileFooter.vue]: though `@screen` only covers min-width

Comment (mindfulme) [reply src/components/MobileFooter/MobileFooter.vue]: I would rather use multiple ranges like so 
screen shot 2018-10-24 at 01 10 48

**Comment by roma219** [comment]
Open when it looks & behaves like in design and ready for review

**Comment by mindfulme** [comment]
@roma219 
screen shot 2018-10-30 at 02 07 46

File: src/components/MobileFooter/MobileFooter.vue [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+</script>
+
+<style>
+@screen sm {
+    .mobile-footer {
+        display: none;
+    }
+}
+
+@media (max-width: config('screens.sm')) {
```
Comment (roma219) [inline src/components/MobileFooter/MobileFooter.vue]: No need for media queries at all, use tailwind classes and responsive prefixes to hide footer on large screens
https://tailwindcss.com/docs/display
https://tailwindcss.com/docs/responsive-design

File: src/components/MobileFooter/MobileFooter.vue [hunk 7]
```diff
@@ -0,0 +1,82 @@
+<template>
+  <div class="mobile-footer">
+    <div
+      v-for="(item, index) in items"
+      :key="index"
+      :style="{ width: itemWidth + '%' }"
+      :class="activeIndex === index ? 'active' : ''"
```
Comment (roma219) [inline src/components/MobileFooter/MobileFooter.vue]: better write it like this:
`:class="{'active': activeIndex === index }"`

File: src/components/MobileFooter/MobileFooter.vue (hunk 6 above)
Comment (roma219) [inline src/components/MobileFooter/MobileFooter.vue]: No need for media queries at all, use tailwind to hide it on large screens:
https://tailwindcss.com/docs/responsive-design
https://tailwindcss.com/docs/display

**Review by roma219 — CHANGES_REQUESTED** [review]
Needs changes, see comments

Also:
1) Should be no line-break
image
2) Active tab text & icon colors should be primary
3) Still not quite like design...:
image
image
4) Use Markets, Orders and Account as test tabs w corresponding icons

**Comment by mindfulme** [comment]
ok, now only need the icons

**Comment by mindfulme** [comment]
@roma219 should be ok, but though there were a small difference regarding a gap in the mockups 
screen shot 2018-10-31 at 00 45 02
screen shot 2018-10-31 at 00 44 54

File: src/components/MobileFooter/MobileFooter.vue [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+.mobile-footer {
+    display: flex;
+    position: fixed;
+    bottom: 0;
+    width:100%;
+    text-align: center;
+    color:config('colors.text-primary');
+}
+
+@screen sm {
```
Comment (roma219) [inline src/components/MobileFooter/MobileFooter.vue]: no media queries

**Review by roma219 — CHANGES_REQUESTED** [review]
remove media querie

File: tailwind.js [hunk 9]
```diff
@@ -185,7 +187,7 @@ module.exports = {
     'sm': '576px',
     'md': '768px',
     'lg': '992px',
-    'xl': '1200px'
+    'xl': '1200px',
```
Comment (roma219) [inline tailwind.js]: trailing comma

**Review by roma219 — CHANGES_REQUESTED** [review]
Remove trailing comma and it's good to go

File: tailwind.js (hunk 9 above)
Comment (mindfulme) [reply tailwind.js]: probably did not include in the commit

Comment (mindfulme) [reply tailwind.js]: got it

**Review by roma219 — APPROVED** [review]

**Pull request merged**